src/Core/Output/TypedDataDumper.cs

```diff
@@ -41,11 +41,99 @@ public TypedDataDumper(EndianImageReader rdr, uint cbSize, Formatter stm)
         public void VisitArray(ArrayType at)
         {
             var addrEnd = rdr.Address + cbSize;
-            for (int i = 0; at.IsUnbounded || i < at.Length; ++i)
+
+            if (at.ElementType.Domain < Domain.Composite) 
             {
-                if (!rdr.IsValid || addrEnd <= rdr.Address)
-                    return;
-                at.ElementType.Accept(this);
+                int offset = 0;
```

Comment:
There appears to be a lot of duplication here; could this code and the code in `VisitPrimitiveType` be harmonized?